kore/src/Kore/Equation/Equation.hs

```diff
@@ -168,9 +168,8 @@ toTermLike sort equation
                         argumentTerm
                     )
                 )
-                ( TermLike.mkAnd
-                    (TermLike.mkEquals sort left right)
-                    ensures'
+                ( TermLike.mkEquals sort left $
```

Comment:
This is used for unparsing, right? If so, I think the changes to this function are premature: until the frontend changes its equation encoding we shouldn't make changes here. Not sure if this is the best solution, but we could add a `toTermLikeTODO` function with the new encoding which will replace this once the frontend changes.